inbm-lib/inbm_common_lib/utility.py

```diff
@@ -180,11 +180,11 @@ def canonicalize_uri(url: str) -> CanonicalUri:
 
     WARNING: a string with no forward slashes will have a scheme added. e.g., 'a' -> 'https://a'
 
-    @param url: the url
+    @param url: URL
     @return canonicalized version"""
 
     if url and URL_NULL_CHAR in url:
-        raise UrlSecurityException("Unsafe characters detected in the url. Cannot process the request.")
+        raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")
 
     return CanonicalUri(value=url_normalize.url_normalize(url))
```

Comment:
**Suggestion:** Consider validating the URL before canonicalization to ensure it has a valid format. This can prevent adding schemes to invalid URLs which might not be intended behavior. [security]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    if url and URL_NULL_CHAR in url:
        raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")

    # Validate the URL format before canonicalization
    if not validate_url_format(url):
        raise UrlSecurityException("Invalid URL format.")

    return CanonicalUri(value=url_normalize.url_normalize(url))
```